hedera-node/src/main/java/com/hedera/services/txns/validation/ContextOptionValidator.java

```diff
@@ -133,7 +136,27 @@ public JKey attemptDecodeOrThrow(final Key k) {
 			throw new InvalidTransactionException(ResponseCodeEnum.BAD_ENCODING);
 		}
 	}
-	
+
+	/**
+	 * <p>Tries to decode a key, using {@link JKey#mapKey(Key)}.</p>
+	 * <p>If the decoding fails - logs the error and throws an {@link InvalidTransactionException}.</p>
+	 *
+	 * @param key
+	 * 		the key, subject of the decoding operation
+	 * @param topicID
+	 * 		the topic's id, used for composing the logged error message
+	 * @return
+	 *     {@link JKey}
+	 */
+	@Override
+	public JKey attemptDecodeOrThrow(final Key key, TopicID topicID) {
+		try {
+			return JKey.mapKey(key);
+		} catch (DecoderException e) {
+			log.error("Decoder exception updating topic {}. ", topicID, e);
```

Comment:
I think that this method **could** and will be used in other places. Hardcoding the message IMO is not a good approach. We should use the already introduced `attemptDecodeOrThrow` method instead of adding new implementation **specific** to the Topics